main/ZgatewayRF.ino

```diff
@@ -264,28 +366,50 @@ void XtoRF(const char* topicOri, JsonObject& RFdata) { // json object decoding
 }
 #  endif
 
-int receiveInterupt = -1;
-
+/**
+ * @brief Disables the RF receiver.
+ *
+ * This function disables the RF receiver by calling the disableReceive method
+ * on the mySwitch object. It also logs a trace message indicating that the RF
+ * receiver has been disabled, along with the GPIO pin number used for the RF
+ * receiver.
+ *
+ * @note THIS SEEMS LIKE A DEAD CODE. THE FUNCTION IS NOT CALLED ANYWHERE.
+ */
 void disableRFReceive() {
-  Log.trace(F("disableRFReceive %d" CR), receiveInterupt);
+  Log.trace(F("[RF] disable RFReceive %d" CR), RF_RECEIVER_GPIO);
   mySwitch.disableReceive();
 }
 
+/**
+ * @brief Enables the RF receiver and optionally the RF transmitter.
+ *
+ * This function initializes the RF receiver on the specified GPIO pin and, if not disabled, 
+ * initializes the RF transmitter on the specified GPIO pin. It also sets the RF frequency 
+ * and logs the configuration details.
+ *
+ * @param rfFrequency The frequency for the RF communication in MHz. Default is RFConfig.frequency.
+ * @param rfReceiverGPIO The GPIO pin number for the RF receiver. Default is RF_RECEIVER_GPIO.
+ * @param rfEmitterGPIO The GPIO pin number for the RF transmitter. Default is RF_EMITTER_GPIO.
+ *
+ * @note If RF_DISABLE_TRANSMIT is defined, the RF transmitter will be disabled.
+ */
 void enableRFReceive() {
-  Log.notice(F("Enable RF Receiver: %FMhz" CR), RFConfig.frequency);
-  //RF init parameters
-  Log.notice(F("RF_EMITTER_GPIO: %d " CR), RF_EMITTER_GPIO);
-  Log.notice(F("RF_RECEIVER_GPIO: %d " CR), RF_RECEIVER_GPIO);
+  float rfFrequency = RFConfig.frequency;
+  int rfReceiverGPIO = RF_RECEIVER_GPIO;
+  int rfEmitterGPIO = RF_EMITTER_GPIO;
```

Comment:
Is there a particular reason for not using directly the macros inside the logs and avoid adding new variables ?

Reply:
True! It's a leftover from a refactor I was doing on a separate branch. I was trying to understand the flow of this function and discovered that some variables were defined outside the function, while others were present as global defines... Initially, I wanted to fix this issue by creating a function with encapsulated variables and passing the parameters as inputs. However, I never actually changed the function signature! What should I do? Should I update the function signature, or replace them back as magic numbers within the function?

Reply:
I update the function signature